### DISCUSSION TITLE
[Droid] System.Reflection.TargetInvocationException when updating from Uno 4.3.6 to 4.4.13

### DISCUSSION THREAD

**Discussion by TopperDEL** [unoplatform/uno#9468] [Q&A]
### Current behavior

I just update my app from Uno 4.3.6 to 4.4.13. It starts without problems in 4.3.6, but crashes after updating to 4.4.13 with the following expetion:

**System.Reflection.TargetInvocationException:** 'Exception has been thrown by the target of an invocation.'

[mono-rt] [ERROR] FATAL UNHANDLED EXCEPTION: System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation.
[mono-rt]  ---> System.MissingMethodException: Method not found: void Uno.Helpers.DrawableHelper.SetDrawableResolver(System.Func`2<string, int>)
[mono-rt]    at MyApp.App..ctor() in D:\Eigene Daten\Projekte\MyApp\Source\MyApp\MyApp.Uno\MyApp.Uno.Shared\App.xaml.cs:line 19
[mono-rt]    at MyApp.Droid.Application.<>c.<.ctor>b__0_0() in D:\Eigene Daten\Projekte\MyApp\Source\MyApp\MyApp.Uno\MyApp.Mobile\Android\Main.Android.cs:line 26
[mono-rt]    at Microsoft.UI.Xaml.NativeApplication..ctor(AppBuilder appBuilder, IntPtr javaReference, JniHandleOwnership transfer) in C:\a\1\s\src\Uno.UI\UI\Xaml\NativeApplication.cs:line 37
[mono-rt]    at MyApp.Droid.Application..ctor(IntPtr javaReference, JniHandleOwnership transfer) in D:\Eigene Daten\Projekte\MyApp\Source\MyApp\MyApp.Uno\MyApp.Mobile\Android\Main.Android.cs:line 26
[mono-rt]    at System.Reflection.RuntimeConstructorInfo.InternalInvoke(Object obj, Object[] parameters, Boolean wrapExceptions)
[mono-rt]    --- End of inner exception stack trace ---
[mono-rt]    at System.Reflection.RuntimeConstructorInfo.InternalInvoke(Object obj, Object[] parameters, Boolean wrapExceptions)
[mono-rt]    at System.Reflection.RuntimeConstructorInfo.DoInvoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
[mono-rt]    at System.Reflection.RuntimeConstructorInfo.Invoke(BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
[mono-rt]    at System.Reflection.ConstructorInfo.Invoke(Object[] parameters)
[mono-rt]    at Java.Interop.TypeManager.CreateProxy(Type type, IntPtr handle, JniHandleOwnership transfer) in /Users/runner/work/1/s/xamarin-android/src/Mono.Android/Java.Interop/TypeManager.cs:line 337
[mono-rt]    at Java.Interop.TypeManager.CreateInstance(IntPtr handle, JniHandleOwnership transfer, Type targetType) in /Users/runner/work/1/s/xamarin-android/src/Mono.Android/Java.Interop/TypeManager.cs:line 312
[mono-rt]    at Java.Lang.Object.GetObject(IntPtr handle, JniHandleOwnership transfer, Type type) in /Users/runner/work/1/s/xamarin-android/src/Mono.Android/Java.Lang/Object.cs:line 304
[mono-rt]    at Java.Lang.Object._GetObject[Application](IntPtr handle, JniHandleOwnership transfer) in /Users/runner/work/1/s/xamarin-android/src/Mono.Android/Java.Lang/Object.cs:line 290
[mono-rt]    at Java.Lang.Object.GetObject[Application](IntPtr handle, JniHandleOwnership transfer) in /Users/runner/work/1/s/xamarin-android/src/Mono.Android/Java.Lang/Object.cs:line 281
[mono-rt]    at Java.Lang.Object.GetObject[Application](IntPtr jnienv, IntPtr handle, JniHandleOwnership transfer) in /Users/runner/work/1/s/xamarin-android/src/Mono.Android/Java.Lang/Object.cs:line 275
[mono-rt]    at Android.App.Application.n_OnCreate(IntPtr jnienv, IntPtr native__this) in /Users/runner/work/1/s/xamarin-android/src/Mono.Android/obj/Release/net6.0/android-31/mcw/Android.App.Application.cs:line 1071
[mono-rt]    at Android.Runtime.JNINativeWrapper.Wrap_JniMarshal_PP_V(_JniMarshal_PP_V callback, IntPtr jnienv, IntPtr klazz) in /Users/runner/work/1/s/xamarin-android/src/Mono.Android/Android.Runtime/JNINativeWrapper.g.cs:line 22

### Expected behavior

The update should not lead to that error, the app starts like before.

### Workaround

None yet

### Affected platforms

Android

### IDE

Visual Studio 2022

### IDE version

17.3

**Comment by TopperDEL** [comment]
Updating to 4.5.0-dev.453 leads to the same issue.

**Reply by jeromelaban** [reply]
Thanks for the report. This generally means that there's a discrepancy in the versions of Uno that being used in the app, or that visual studio is not publishing the right versions of Uno in the app. Make sure to align all Uno.UI package versions, to clean the bin/obj folders and uninstall the app for the "fast deploy" feature to work properly again.

**Reply by TopperDEL** [reply]
I can confirm it's working now! Thank you!
I've uninstalled the app from the device and removed bin and obj folders.
